src/bindings/python/flux/job/Jobspec.py

```diff
@@ -541,6 +541,8 @@ def getattr(self, key):
         Raises KeyError if a component of key does not exist.
         """
         if not key.startswith("attributes."):
+            if not key.startswith(("user.", "system.")):
```

Comment:
I can see why you also modified `getattr()`, but this seems more dangerous. Since there isn't a "user friendliness" argument for updating `getattr()`, and there is no equivalent `--getattr` command line option which could cause confusion, I wonder if we should just leave getattr alone (it wasn't part of the original bug report)

I'm really on the fence about this one, which is why I think it might be better to not change the behavior.

If nothing else, the documentation here should be updated.

Reply:
good point.  I did waflfe on it and changed it mostly for ease of the testing.  I'll revert it.  We can revisit at a later time if necessary.

Reply:
It does seem weird that a user could do `jobspec.setattr("foo")` and then `jobspec.getattr("foo")` would raise `KeyError`, but like you said we can deal with that if it comes up.